src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -591,7 +591,32 @@ private function getAdjacentRecordID($offset)
 
         $map = $list->limit($limit, $limitOffset)->column('ID');
         $index = array_search($this->record->ID, $map ?? []);
-        return isset($map[$index+$offset]) ? $map[$index+$offset] : false;
+        $position = $index + $offset;
+        $page = $currentPage;
+        $lastPosition = count($map) - 1;
+        $hasMorePages = $this->getNumPages($gridField) > $currentPage;
+
+        if ($position === 0 && $currentPage > 1) {
+            $page = $currentPage - 1;
+        } elseif ($position === $lastPosition && $hasMorePages) {
```

Comment:
```suggestion
        } elseif ($hasMorePages && $position >= $itemsPerPage + 1) {
```
`$position` starts at 0, and ends at `$itemsPerPage + 1`.  The total number of items in that range is `$itemsPerPage + 2` (aka the same as `$limit` or `count($map)`). This means if `$position` is equal to `$itemsPerPage +1` then it is the last item in map, i.e. it is the first item on the _next_ page. `>=` isn't strictly necessary (`===` would also work), but it doesn't hurt to protect ourselves against unexpected edge cases.

We can ignore the case where the number of items in a page is _less_ than `$itemsPerPage` because in that case we're on the last page, so `$hasMorePages` will be false anyway.

Reply:
DONE